BuildScript.Nuke/Components/IDependDB.cs

```diff
@@ -0,0 +1,173 @@
+              Log.Information($"  {nameof(DependDBProjectImportNotificationMailAddress)}: '{DependDBProjectImportNotificationMailAddress}'");
+              Log.Information($"  {nameof(DependDBProjectImportNotificationMailAddressCc)}: '{DependDBProjectImportNotificationMailAddressCc}'");
+              Log.Information($"  {nameof(VcsUrlTemplate)}: '{VcsUrlTemplate}'");
+              Log.Information($"  {nameof(DependDBTargetFramework)}: '{DependDBTargetFramework}'");
+              Log.Information($"  {nameof(DependDBTrackedReferences)}: [{string.Join(", ", trackedReferences.Select(e => $"'{e}'"))}]");
+              Log.Information($"  {nameof(DependDBNuGetSources)}: [{string.Join(", ", nugetSources.Select(e => $"'{e}'"))}]");
+
+              var arguments = new Arguments()
+                      .Add("--project-name={value}", DependDBProjectName)
+                      .Add("--project-branch={value}", DependDBProjectBranch)
+                      .Add("--project-version={value}", projectVersion)
+                      .Add("--retention-time={value}", retentionTime)
+                      .Add("--notification-address-to={value}", DependDBProjectImportNotificationMailAddress)
+                      .Add("--temp-directory={value}", TempFolder)
+                      .Add("--processor-output-directory={value}", outputFolder)
+                      .Add("--preprocessor-output-file={value}", buildProcessorFile)
+                      .Add("--source-directory-root={value}", Solution.Directory)
+                      .Add("--source-control-template={value}", VcsUrlTemplate)
+                      .Add("--own-packages-directory={value}", ownPackagesDirectory)
+                      .Add("--target-framework={value}", DependDBTargetFramework);
+
+              if (DependDBProjectImportNotificationMailAddressCc != null)
+                  arguments = arguments.Add("--notification-address-cc={value}", DependDBProjectImportNotificationMailAddressCc);
+
+              if (DependDBNugetConfig != null)
+                  arguments = arguments.Add("--nuget-config={value}", MakeAbsolutePath(DependDBNugetConfig));
+
+              foreach (var trackedReference in trackedReferences)
+                  arguments = arguments.Add("--analzyer-tracked-reference={value}", trackedReference);
+
+              foreach (var nugetSource in nugetSources)
+                  arguments = arguments.Add("--additional-nuget-source={value}", nugetSource);
+
+              var argumentStringHandler = new ArgumentStringHandler(1, 0, out var _);
+              argumentStringHandler.AppendLiteral(arguments.RenderForOutput());
+
+              NugetPreProcessor(
+                      argumentStringHandler,
+                      workingDirectory: Solution.Directory);
+
+              BuildProcessor(new Arguments().Add(buildProcessorFile).RenderForExecution());
+
+
+              string MakeAbsolutePath (string value)
+              {
+                  if (value.StartsWith("~/"))
```

Comment:
The thing is that we can't have absolute paths in the configuration files that are located in the repo because the checkout directory might change. Relative paths might be interpreted incorrectly using `Path.CurrentDirectory` so I decided to use the ASP.NET pattern `~/` to indicate the application root (-> the root of the repo).